testing/simulated/pectra_withdrawal_test.go

```diff
@@ -0,0 +1,547 @@
+		s.Require().Equal(int64(2)*nextBlockHeight, nextBlockTime.Unix())
+
+		validators, err := s.TestNode.APIBackend.FilteredValidators(beaconmath.Slot(nextBlockHeight), nil, nil)
+		s.Require().NoError(err)
+		s.Require().Len(validators, 2)
+		s.Require().Equal(validator.ExitedUnslashed.String(), validators[1].Status)
+
+		// The validator should withdrawable at Epoch 6 since hard fork has not occurred.
+		s.Require().Equal("6", validators[1].Validator.WithdrawableEpoch)
+		nextBlockHeight++
+	}
+	// [Slot/Epoch 6] Move the chain by 1. This block activates the hard fork. No withdrawal delay is expected.
+	{
+		s.LogBuffer.Reset()
+
+		_, _, nextBlockTime = s.MoveChainToHeight(s.T(), nextBlockHeight, 1, blsSigner, nextBlockTime)
+		s.Require().Equal(int64(2)*nextBlockHeight, nextBlockTime.Unix())
+
+		validators, err := s.TestNode.APIBackend.FilteredValidators(beaconmath.Slot(nextBlockHeight), nil, nil)
+		s.Require().NoError(err)
+		s.Require().Len(validators, 2)
+		s.Require().Equal(validator.WithdrawalPossible.String(), validators[1].Status)
+
+		// The validator should withdrawable at Epoch 6 since hard fork has not occurred.
+		s.Require().Equal("6", validators[1].Validator.WithdrawableEpoch)
+
+		// Confirm the fork was activated
+		s.Require().Contains(s.LogBuffer.String(), "✅  welcome to the electra (0x05000000) fork! 🎉")
+
+		// Confirm the balance change on EL
+		previousBlockBalance, err := s.TestNode.ContractBackend.BalanceAt(s.CtxApp, senderAddress, big.NewInt(nextBlockHeight-1))
+		s.Require().NoError(err)
+		currentBalance, err := s.TestNode.ContractBackend.BalanceAt(s.CtxApp, senderAddress, big.NewInt(nextBlockHeight))
+		s.Require().NoError(err)
+		depositAmountWei := big.NewInt(0).Mul(big.NewInt(int64(depositAmount)), big.NewInt(1e9))
+		expectedBalance := big.NewInt(0).Add(previousBlockBalance, depositAmountWei)
+		s.Require().Equal(expectedBalance, currentBalance)
+		nextBlockHeight++
+	}
+	// [Slot/Epoch 7] Move the chain by 1. The effective balance is now 0 and the withdrawal is complete
+	{
+		s.LogBuffer.Reset()
+
+		_, _, nextBlockTime = s.MoveChainToHeight(s.T(), nextBlockHeight, 1, blsSigner, nextBlockTime)
+		s.Require().Equal(int64(2)*nextBlockHeight, nextBlockTime.Unix())
+
+		validators, err := s.TestNode.APIBackend.FilteredValidators(beaconmath.Slot(nextBlockHeight), nil, nil)
+		s.Require().NoError(err)
+		s.Require().Len(validators, 2)
+		s.Require().Equal(validator.WithdrawalDone.String(), validators[1].Status)
+		nextBlockHeight++
+	}
+}
+
+// Verifies that excess balance is not withdrawn accidentally if a validator has multiple sources of withdrawals.
+// Both the partial withdrawal and the excess balance withdrawal will occur simultaneously in a block.
+// Multiple deposits are sent so that Consensus Layer has a higher balance when the withdrawal request is processed.
+// This also served as a PoC for a now patched bug.
```

Comment:
<3 nit: I like that we indicate that some tests are there as regression, but please consider referencing the commit where fix happened